src/IndexSetAnalysis.h

```diff
@@ -8,31 +8,98 @@
 
 /***************************************************************************
  *
- * @file AutoIndex.h
+ * @file IndexSetAnalysis.h
  *
- * Defines classes to automatically compute the minimal indexes for a table
+ * Computes indexes for relations in a translation unit
  *
  ***************************************************************************/
 
 #pragma once
 
-#include "MaxMatching.h"
+#include "RamAnalysis.h"
+#include "RamRelation.h"
 #include "RamTypes.h"
+#include "Util.h"
 
+#include <cstring>
+#include <functional>
 #include <iostream>
 #include <limits>
+#include <map>
+#include <queue>
```

Comment:
std::queue isn't used in this header file, only in the cpp. Move the include to the cp

Reply:
I have removed even more unnecessary includes.